book/src/app.md

```diff
@@ -1,54 +1,33 @@
 # Application
 
-`App` define the type of note (e.g. XAN, ETH, BTC). Each application is identified by an address `appAddress` (the same way as user address identifies a user) and has its own VP `AppVP`.
+### Application logic
+Application VP is the defining structure of an application that contains the application logic. 
+Every time the application state is about to change, the application VP is called to validate the state transition.
+If the application VP doesn't consider a state transition valid, the state will not be changed.
 
-### Application VP
-Each application has its own [validity predicate](./validity-predicates.md) `AppVP` that defines the conditions on which the application can be sent/received/etc (e.g. whitelist VP that only allows using the application a specified set of users). As with other VPs, `AppVP` checks that input and output notes of the tx satisfy certain constraints.
-It is required that the `AppVP` of the applications involved in a tx evaluated to `true`.
+An application VP might require validity of other VPs, enforcing the VP hierarchy with the application VP being the main VP checked.
 
-In Taiga, VPs are shielded, so instead of showing that `AppVP` evaluates to `true` publicly, a ZK proof is created. To make sure that `AppVP`  evaluates to `true`, an observer can verify the proof (using the verifier key):
+#### Application address
+Each application has an address that identifies the application. The application address is derived from its application VP (or VPs, if there is more than one).
```

Comment:
Do we have more than one application VP at the moment? I think it would help to remove "(or VPs, if there is more than one)" since it brings (to me) some questions. Or if we want to talk about how to have multiple applications, we can have a section for it.

Reply:
Is the address of an application its verifying key?

Reply:
The address is derived from the application verifying key, but doesn't equal it. I think mentioning verifying key explicitly is most of the time unnecessary for the overview, this is a detail that doesn't help to understand the architecture more, but adds more information. It would be more appropriate to describe in details in the spec

Reply:
About multiple VPs: to me it was more about available options, but I see how it can be confusing for the reader. I'll remove it